pkg/types/change/change.proto

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2019-present Open Networking Foundation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+package onos.config.change;
+
+import "google/protobuf/timestamp.proto";
+import "gogoproto/gogo.proto";
+
+// Status is the status of a configuration object
+enum Status {
+    // PENDING indicates the configuration is waiting to be applied
+    PENDING = 0;
+
+    // APPLYING indicates the configuration is in the process of being applied
+    APPLYING = 1;
+
+    // SUCCEEDED indicates the configuration was applied successfully
+    SUCCEEDED = 2;
+
+    // FAILED indicates the configuration failed
+    FAILED = 3;
+}
+
+// Reason is a failure reason
+enum Reason {
+    // ERROR indicates an error occurred
+    ERROR = 0;
+
+    // UNAVAILABLE indicates the device was unavailable
+    UNAVAILABLE = 1;
+}
+
+// Change represents a configuration change to a single device
+message Change {
+    // The identifier of the change
+    string id = 1 [(gogoproto.customname) = "ID", (gogoproto.casttype) = "ID"];
```

Comment:
Are you thinking of an auto-generated hash for this? I don't think the user will be able to set it, as it's only created by the Network service.